### PR TITLE
[Localization] Ability Bar can now be localized. In a way that pokemonName abilityNa…

### PR DESCRIPTION
**Pull request by CodeTappert**
…me and passive string can be ordered freely

## What are the changes?

The ability bar can now be localized and also wraps automatically if the text is too long

## Why am I doing these changes?

Because the way it was before only made sense in english (well at least it didnt in german)

## What did change?

Made it localizable with the ability to place abilityName, pokemonName and passive freely

### Screenshots/Videos

- Ability that is no passive
https://github.com/pagefaultgames/pokerogue/assets/38758606/2427362e-d7bf-4afe-89a4-6ba0f47275cc

- Ability that is a passive
https://github.com/pagefaultgames/pokerogue/assets/38758606/dd39713d-56ac-4a25-a832-dddbe523ef69

English Text remains unchanged to how it was before

## How to test the changes?

Use any pokemon with a ability that triggers such a box (like kyogre or groudon)

## Checklist
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Comment by CodeTappert** [comment]
Currently filling the text at the top.

File: src/ui/ability-bar.ts [hunk 1]
```diff
@@ -26,21 +26,17 @@ export default class AbilityBar extends Phaser.GameObjects.Container {
...
+    this.add(this.abilityBarText);
 
     this.setVisible(false);
     this.shown = false;
   }
 
   showAbility(pokemon: Pokemon, passive: boolean = false): void {
-    this.pokemonNameText.setText(`${pokemon.name}'s${passive ? " Passive" : ""}`);
-    this.abilityNameText.setText((!passive ? pokemon.getAbility() : pokemon.getPassiveAbility()).name);
+    this.abilityBarText.setText(`${i18next.t("fightUiHandler:abilityFlyInText", { pokemonName: pokemon.name, passive: passive ? i18next.t("fightUiHandler:passive") : "", abilityName: !passive ?  pokemon.getAbility().name : pokemon.getPassiveAbility().name })}`);
```
Comment (torranx) [inline src/ui/ability-bar.ts:39]: can't we just add the space here?

Comment (CodeTappert) [reply src/ui/ability-bar.ts:39]: We could. But i dont see the need todo so. There are plently of localization strings that start or end with a " "

File: src/locales/ko/fight-ui-handler.ts [hunk 2]
```diff
@@ -4,4 +4,6 @@ export const fightUiHandler: SimpleTranslationEntries = {
   "pp": "PP",
   "power": "위력",
   "accuracy": "명중률",
+  "abilityFlyInText": " {{pokemonName}}'s {{passive}}{{abilityName}}",
+  "passive": "Passive ", // The space at the end is important
```
Comment (returntoice) [inline src/locales/ko/fight-ui-handler.ts:8]:
```suggestion
  "abilityFlyInText": " {{pokemonName}}의 {{passive}}{{abilityName}}",
  "passive": "패시브 ", // The space at the end is important
```
Korean translation added

**Review by returntoice — CHANGES_REQUESTED** [review]

File: src/locales/de/fight-ui-handler.ts [hunk 3]
```diff
@@ -4,4 +4,6 @@ export const fightUiHandler: SimpleTranslationEntries = {
   "pp": "AP",
   "power": "Stärke",
   "accuracy": "Genauigkeit",
+  "abilityFlyInText": "{{passive}}{{abilityName}} von {{pokemonName}} wirkt",
```
Comment (CodeTappert) [inline src/locales/de/fight-ui-handler.ts:7]:
```suggestion
  "abilityFlyInText": "{{passive}}{{abilityName}} von {{pokemonName}} wirkt.",
```

File: src/locales/de/fight-ui-handler.ts [hunk 4]
```diff
@@ -4,4 +4,6 @@ export const fightUiHandler: SimpleTranslationEntries = {
   "pp": "AP",
   "power": "Stärke",
   "accuracy": "Genauigkeit",
+  "abilityFlyInText": "{{passive}}{{abilityName}} von {{pokemonName}} wirkt.",
```
Comment (CodeTappert) [inline src/locales/de/fight-ui-handler.ts:7]:
```suggestion
  "abilityFlyInText": "{{passive}}{{abilityName}} von {{pokemonName}} wirkt!",
```

**Comment by flx-sta** [comment]
image
image

**Review by flx-sta — APPROVED** [review]
### lgtm 🚀 
image
image

If both have it:


https://github.com/pagefaultgames/pokerogue/assets/50131232/87cd384f-ba79-4c4f-a147-8c6cdef8b70d

**Review by Tempo-anon — APPROVED** [review]

**Comment by DustinLin** [comment]
Just another sample of traditional chinese, seems to work as intended
https://github.com/pagefaultgames/pokerogue/assets/39450497/b94b0a04-abc4-49f2-b431-9b2a5ae082ad

**Review by returntoice — APPROVED** [review]

**Comment by CarsonVB** [comment]
Due to the new handling of the ability bar text, the formatting is incorrect in the legacy UI
image
Previously the Pokemon's name appeared on the top of the box, where the text currently is, then the ability name would appear inside the box, like this
image

**Pull request merged**